src/ui/public/state_management/state_storage/hashing_store.js

```diff
@@ -1,20 +1,13 @@
-import angular from 'angular';
-import { sortBy } from 'lodash';
-import { Sha256 } from 'ui/crypto';
-
-import StubBrowserStorage from 'test_utils/stub_browser_storage';
-import { LazyLruStore } from './lazy_lru_store';
-
 /**
  *  The HashingStore is a wrapper around a browser store object
  *  that hashes the items added to it and stores them by their
  *  hash. This hash is then returned so that the item can be received
  *  at a later time.
  */
 class HashingStore {
-  constructor({ store, createHash, maxItems } = {}) {
-    this._store = store || window.sessionStorage;
-    if (createHash) this._createHash = createHash;
+  constructor(createStorageHash, store) {
```

Comment:
I think config objects are great when dealing with a ton of arguments (more than 3), since they add more meaning for each argument when invoking the method.

On the other hand, there's also some meaning in the order in which arguments are passed (in this case, the createStoreHash function is used to set items in the store, so it comes first). And config objects require additional syntax in their creation and in their consumption.

Since this constructor only needs two arguments, I don't think this is an appropriate case for a config object. The readability benefits of the config object through named parameters don't exist any more and the additional syntax required to use the config object becomes noise, in my opinion.

Also, personally, config objects stand out to me as yellow flags that indicate there may be a way to simplify the code and user fewer arguments. So by removing the config object here, I've removed that yellow flag for myself.